cpp/ql/test/query-tests/Security/CWE/CWE-119/semmle/tests/tests.cpp

```diff
@@ -46,10 +46,10 @@ void test3()
 	smallbuffer = new char[10];
 	bigbuffer = new char[20];
 
-	memcpy(bigbuffer, smallbuffer, sizeof(smallbuffer)); // GOOD
-	memcpy(bigbuffer, smallbuffer, sizeof(bigbuffer)); // BAD: over-read [NOT DETECTED]
-	memcpy(smallbuffer, bigbuffer, sizeof(smallbuffer)); // GOOD
-	memcpy(smallbuffer, bigbuffer, sizeof(bigbuffer)); // BAD: over-write [NOT DETECTED]
+	memcpy(bigbuffer, smallbuffer, 10); // GOOD
+	memcpy(bigbuffer, smallbuffer, 20); // BAD: over-read
+	memcpy(smallbuffer, bigbuffer, 10); // GOOD
+	memcpy(smallbuffer, bigbuffer, 20); // BAD: over-write
```

Comment:
We would prefer the following:
```suggestion
	memcpy(bigbuffer, smallbuffer, sizeof(char[10])); // GOOD
	memcpy(bigbuffer, smallbuffer, sizeof(char[20])); // BAD: over-read
	memcpy(smallbuffer, bigbuffer, sizeof(char[10])); // GOOD
	memcpy(smallbuffer, bigbuffer, sizeof(char[20])); // BAD: over-write
```